src/puppetlabs/ezbake/dependency_utils.clj

```diff
@@ -30,6 +30,47 @@
       (classpath/merge-versions-from-managed-coords
        (:managed-dependencies lein-project))))
 
+(defn snapshot-version?
+  [version]
+  ;; dependencies managed by lein-parent have nil versions so guard against that
+  (boolean (and version (re-find #"SNAPSHOT" version))))
+
+(defn get-full-snapshot-version
+  [lein-project coords]
+  (let [full-version (-> (aether/resolve-artifacts
+                           :coordinates [coords]
+                           :repositories (:repositories lein-project)
+                           :mirrors (:mirrors lein-project)
+                           :local-repo (:local-repo lein-project))
+                       first
+                       meta
+                       :result
+                       .getArtifact
+                       .getVersion)]
+    (if-not (re-find #"SNAPSHOT$" full-version)
+      full-version
+      (throw
+        (Exception.
```

Comment:
total nitpick, but it would be good to throw something slightly more specific here, like IllegalArgumentException or IllegalStateException maybe?